src/Driver/solver_configs.jl

```diff
@@ -118,17 +118,23 @@ function SolverConfiguration(
 
         state_auxiliary = restart_auxiliary_state(bl, grid, s_aux, direction)
 
-        dg = DGModel(
-            bl,
-            grid,
-            numerical_flux_first_order,
-            numerical_flux_second_order,
-            numerical_flux_gradient;
-            state_auxiliary = state_auxiliary,
-            direction = direction,
-            diffusion_direction = diffdir,
-            modeldata = modeldata,
-        )
+        if hasproperty(driver_config.config_info, :dg)
```

Comment:
This makes me uncomfortable, but it'll get fixed when #1551 is implemented so I guess it's fine for now.

Reply:
This is me completely hacking in #1551 like functionality. After seeing how SolverTypes work, I might be able to move most of this to the SplitExplicitSolverType for now? Let me know what you think.

Reply:
I think, as part of addressing #1551, we can introduce `SolverConfiguration` constructors specialized on `driver_config.config_type`, as those constructors will basically become wrappers for `solversetup`.